File: LeetCode/Kth Smallest Element in a Sorted Matrix/kth_smallest_element_in_a_sorted_matrix.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        vector<int> v;
        
        for(int i = 0;i<matrix.size();i++)
        {
            for(int j = 0;j<matrix[0].size();j++)
                v.push_back(matrix[i][j]);
        }
        sort(v.begin(),v.end());
        
        return v[k-1];
    }
};

File: LeetCode/Kth Smallest Element in a Sorted Matrix/Kth Smallest Element in a Sorted Matrix.py
def kthSmallest(matrix, k):
  row = k // len(matrix) - 1 
  col = k % len(matrix)
  return matrix[row][col]

# Example 1
# Input: matrix = [[1,5,9],[10,11,13],[12,13,15]], k = 8
# Output: 13
print(kthSmallest([[1,5,9],[10,11,13],[12,13,15]], 8))

# Example 2
# Input: matrix = [[-5]], k = 1
# Output: -5
print(kthSmallest([[-5]], 1))